quint/src/names/base.ts

```diff
@@ -100,6 +100,28 @@ export function copyNames(
   return table
 }
 
+/**
+ * Add namespaces to a definition's `namespaces` field, if it doesn't already
+ * have them on the latest position or in the beginning of its name.
+ *
+ * @param def - The definition to add the namespaces to
+ * @param namespaces - The namespaces to be added
+ *
+ * @returns The definition with the namespaces added
+ */
+export function addNamespacesToDef(def: Definition, namespaces: string[]): Definition {
+  return namespaces.reduce((def, namespace) => {
+    if (
+      (def.namespaces && def.namespaces[def.namespaces?.length - 1] === namespace) ||
+      def.name.startsWith(namespace)
```

Comment:
Oh well, you broke that pretty easily :laughing:. I probably should have mentioned this in the ADR, but this namespace manipulation part is also very tricky. I was hoping I could make it better in the implementation step (now), but there's way too much interaction between the parts involved here for me to be confident to change this single part alone.

Do you think it's ok if I try to fix this scenario after the other parts are in place? I can add a unit test to describe the faulty behavior now, open an issue, and put it in my queue right after the wiring in of flatteners.

One thing I should do right now is to change this to `def.name.startsWith('${namespace}::')` to mitigate the problem a bit. To actually solve it, I think I might need a field like `namespacesAlreadyAddedToTheName` and check that instead of the actual name.

Reply:
I spent some time on this and found still more problems. I'll try to figure it out on Monday with a fresher head.

Reply:
Yeah, we can postpone working on corner cases, if that blocks your overall understanding. Just let's have a follow up issue on that.

Reply:
I think this function is actually way too general for the scope in which I'll need it. Matching prefixes completely here would introduce unnecessary complexity. I have to play around with the full implementation to understand the scope of cases we actually need to take care of here, so I'll do that once the other parts are in place.